firefox-ios/Client/Frontend/ContextualHint/ContextualHintViewController.swift

```diff
@@ -114,9 +114,9 @@ class ContextualHintViewController: UIViewController,
 
     private func performAction() {
         self.viewProvider.sendTelemetryEvent(for: .performAction)
-        self.dismiss(animated: true) {
-            self.onActionTapped?()
-            self.onActionTapped = nil
+        self.dismiss(animated: true) { [weak self] in
```

Comment:
you don't need to use a [weak self] as there is no retain cycle

Reply:
You're correct it shouldn't be necessary here; removed, thanks.